### PR TITLE
Fix semantic operations on type parameters on type aliases

### LINKED ISSUE #3989: Error for findReferences/getDocHighlights/quickInfo on type alias type parameter
Try getting references/occurrences/quickinfo at the following markers:

```TypeScript
type List</*1*/T> = /*2*/T[];
```

```
Description:

getDocumentHighlights('C:\Users\drosen\AppData\Local\Temp\~vs65CE.ts', 7)

StackTrace:

TypeError: Unable to get property 'length' of undefined or null reference
   at getSignatureFromDeclaration (:15933:33)
   at getSymbolDisplayPartsDocumentationAndSymbolKind (:43634:21)
   at getDefinition (:44650:17)
   at getReferencedSymbol (:44877:25)
   at Anonymous function (:44860:33)
   at forEach (:849:17)
   at getReferencesInNode (:44827:21)
   at getReferencedSymbolsForNode (:44644:25)
   at getSemanticDocumentHighlights (:44021:21)
   at getDocumentHighlights (:44005:13)
   at Anonymous function (:47564:17)
   at simpleForwardCall (:47294:9)
   at forwardJSONCall (:47310:13)
   at forwardJSONCall (:47356:13)
   at getDocumentHighlights (:47563:13)
```

### PR DESCRIPTION
**Pull request by yuit**
fix #3989. The fix is tested through updating language service and using it in VS. The PR fix null reference and prevent completions to be shown on type parameter in type alias (similar to type parameter in function/class/interface etc.)

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
Where are the tests for this change?

**Comment by mhegazy** [comment]
change looks good. please add a unit test, and we should be good to go.

**Comment by DanielRosenwasser** [comment]
More than just a unit test - the semantic operations for the type alias that caused the crash should be tested (obviously we had no tests for this to begin with), as well as something that should specifically trigger the cause of the crash (i.e. quick info probably).

**Comment by mhegazy** [comment]
:+1:

File: tests/cases/fourslash/completionListInTypeParameterOfClassExpression1.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿/// <reference path='fourslash.ts'/>
+
+////var C0 = class D</*0*/
+////var C1 = class D</*1*/T> {}
+////var C3 = class D<T, /*2*/
+////var C4 = class D<T, /*3*/U>{}
```
Comment (DanielRosenwasser) [inline tests/cases/fourslash/completionListInTypeParameterOfClassExpression1.ts]: Can you add a case for

```TypeScript
////var C5 = class D<T extends /*4*/>{}
```

File: tests/cases/fourslash/completionListInTypeParameterOfTypeAlias2.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+goTo.marker("0");
+verify.completionListIsEmpty();
+goTo.marker("1");
+verify.completionListContains("V");
+goTo.marker("2");
+verify.not.completionListIsEmpty();
+verify.completionListContains("K");
+verify.completionListContains("V");
+goTo.marker("3");
+verify.not.completionListIsEmpty();
```
Comment (DanielRosenwasser) [inline tests/cases/fourslash/completionListInTypeParameterOfTypeAlias2.ts]: Remove this, you already test for the presence of items below

File: tests/cases/fourslash/quickInfoDisplayPartsTypeParameterInTypeAlias.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+﻿/// <reference path='fourslash.ts'/>
+
+////type /*0*/List</*1*/T> = /*2*/T[]
```
Comment (DanielRosenwasser) [inline tests/cases/fourslash/quickInfoDisplayPartsTypeParameterInTypeAlias.ts]: Can you add one for

```TypeScript
////type /*0*/List2</*1*/T extends string> = /*2*/T[];
```

**Pull request merged**